migrator/data/DataImporter.java

```diff
@@ -520,8 +600,57 @@ public ByteArray get(String originalID) {
             }
         }
 
-        public boolean contains(String originalID) {
-            return get(originalID) != null;
+        public boolean containsMapped(String originalID) {
+            return getMapped(originalID) != null;
+        }
+
+        public void recordIncomplete(String originalID) {
+            ByteArray key = ByteArray.join(Prefix.ID_INCOMPLETE.bytes, encodeOriginalID(originalID));
+            try {
+                storage.put(key.getBytes(), new byte[0]);
+            } catch (RocksDBException e) {
+                throw TypeDBException.of(e);
+            }
+        }
+
+        public boolean isIncomplete(String originalID) {
+            ByteArray key = ByteArray.join(Prefix.ID_INCOMPLETE.bytes, encodeOriginalID(originalID));
+            try {
+                return storage.get(key.getBytes()) != null;
+            } catch (RocksDBException e) {
+                throw TypeDBException.of(e);
+            }
+        }
+
+        public boolean containsIncomplete() {
+            RocksIterator iterator = storage.newIterator();
+            iterator.seek(Prefix.ID_INCOMPLETE.bytes.getBytes());
+            boolean hasIncomplete = iterator.isValid() && ByteArray.of(iterator.key()).hasPrefix(Prefix.ID_INCOMPLETE.bytes);
+            iterator.close();
+            return hasIncomplete;
+        }
+
+        public void deleteIncomplete(String originalID) {
+            ByteArray key = ByteArray.join(Prefix.ID_INCOMPLETE.bytes, encodeOriginalID(originalID));
+            try {
+                storage.delete(key.getBytes());
+            } catch (RocksDBException e) {
+                throw TypeDBException.of(e);
+            }
+        }
+
+        private ByteArray encodeOriginalID(String originalID) {
+            return ByteArray.encodeString(originalID, STRING_ENCODING);
+        }
+
+        enum Prefix {
```

Comment:
we have two keys we need to store into our temporary RocksDB storage. We had to introduce this to replace all-in memory data structures, otherwise we could OOM on large imports!